src/counterfactuals/core_struct.jl

```diff
@@ -67,6 +67,7 @@ function CounterfactualExplanation(
         :invalidation_rate,
         :early_stopping,
     ]
+    @assert predict_label(M, data, x) != target "The factual `x` is already in the target class. No counterfactuals can be generated."
```

Comment:
I'm actually a bit surprised it wasn't there in the first place, I added it to simplify the growing spheres generator algorithm. Does that make sense to you @pat-alt?